src/status_im/contexts/settings/wallet/events.cljs

```
@@ -82,3 +88,81 @@
+        keypair-name (data-store/extract-keypair-name db key-uids-set)]
+    {:db (-> db
+             (update-in [:wallet :accounts] #(data-store/make-accounts-fully-operable % key-uids-set))
+             (update-in [:wallet :keypairs] #(data-store/make-keypairs-fully-operable % key-uids-set)))
+     :fx [[:dispatch
+           [:toasts/upsert
+            {:type  :positive
+             :theme :dark
+             :text  (if (= (count key-uids-to-update) 1)
+                      (i18n/label :t/key-pair-imported-successfully {:name keypair-name})
+                      (i18n/label :t/key-pairs-successfully-imported
+                                  {:count (count key-uids-to-update)}))}]]]}))
+
+(rf/reg-event-fx :wallet/make-keypairs-accounts-fully-operable make-keypairs-accounts-fully-operable)
+
+(rf/reg-fx :effects.connection-string/import-keypair
+ (fn [{:keys [key-uid sha3-pwd keypairs-key-uids connection-string on-success on-fail]}]
+   (let [config-map (.stringify js/JSON
+                                (clj->js
+                                 {:receiverConfig
+                                  {:loggedInKeyUid   key-uid
+                                   :keystorePath     ""
+                                   :password         (security/safe-unmask-data
+                                                      sha3-pwd)
+                                   :keypairsToImport keypairs-key-uids}}))]
+     (-> (native-module/input-connection-string-for-importing-keypairs-keystores
+          connection-string
+          config-map)
+         (promesa/then (fn [res]
+                         (let [error (when (syncing-events/extract-error res)
+                                       (str "generic-error: " res))]
+                           (if-not (some? error)
+                             (on-success)
+                             (on-fail error)))))
+         (promesa/catch #(log/error "error import-keypair/connection-string " %))))))
```

Inside `catch` we should probably call `on-fail` too, otherwise we swallow the error.